projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts

```diff
@@ -62,19 +62,36 @@ export class DataLayerManagerService {
    * You can observe the retrieved layers using one of the manager's Observable properties:
    * [allLayers$], [selectedLayers$], or [availableLayers$].
    */
-  retrieveAll() {
+  retrieveAll(selected?: string[]) {
     this.loading$.next(true);
     merge(...this.dataLayerServices.map((service) => service.retrieve())).subscribe({
       next: (layer) =>
         this.stateSubject.next({
           ...this.stateSubject.value,
           layers: this.mergeService.merge(this.state.layers, layer),
         }),
+
       error: (err) => console.error(err),
       complete: () => {
         this.loading$.next(false);
+        this.selectAll(selected)
       },
     });
+
+  }
+
+  selectAll(selected?: string[]) {
+    selected?.map(id => this.stateSubject.next(
+      produce(this.state, (draft) => {
+        const layer = draft.layers[id];
+        console.log(draft.layers)
+        draft.selected.push(layer.id);
+        layer.selected = true;
+        layer.enabled = true;
+        this.colorService.chooseColorsFromPalette(layer);
+      })
+    ))
```

Comment:
1. Why does this need to duplicate the code from the select method? can it just call the method instead?
2. If you are not using the return value of `map`, use `forEach` or a `for...of` loop instead.